### PR TITLE
Add support for ip route add blackhole

### PR DESCRIPTION
**Pull request by mhio**
- Maps to adding the `-blackhole` option on mac `route`
- show moves blackhole entries to the bottom
- show ipv4 processing moved into helper function so blackhole can call it
- add tests for add/delete/show blackhole

For #36 but needs some testing as it modifies the way `route add` args are parsed and how `route show` processes ipv4 addresses so I've made some changes that looked ok at first pass but I'm sure there will be side effects I'm not aware of.

### CODE REVIEW

File: src/ip.py [hunk 1]
```diff
@@ -123,7 +124,7 @@ def do_help_neigh():
 def do_route(argv,af):
   if not argv:
     return do_route_list(af)
-  elif 'add'.startswith(argv[0]) and len(argv) >= 4:
+  elif 'add'.startswith(argv[0]) and len(argv) >= 3:
```
Comment (mhio) [inline src/ip.py:127]: I'm not sure what this will open up arg parsing wise?

File: src/ip.py [hunk 2]
```diff
@@ -185,45 +190,69 @@ def do_route_list(af):
...
         if re.match("link.+",gw):
-          print(addr + '/' + str(netmask)+ ' dev ' + dev + '  scope link')
+          print(cidr + ' dev ' + dev + '  scope link')
         else:
-          print(addr + '/' + str(netmask) + ' via ' + gw + ' dev ' + dev)
+          print(cidr + ' via ' + gw + ' dev ' + dev)
+  for b in blackholes:
+    target = b[0]
+    if af == 6:
+      target = re.sub('%[^ ]+/','/',target)
```
Comment (mhio) [inline src/ip.py:207]: This looks like all that needs to happen to process ipv6?

File: src/ip.py [hunk 3]
```diff
@@ -185,45 +190,69 @@ def do_route_list(af):
...
+  elif dots == 0:
+    addr = addr + '.0.0.0'
+  return addr + '/' + str(netmask)
+
+
 def do_route_add(argv,af):
   target=argv[0]
   via=argv[1]
-  gw=argv[2]
+  options=""
```
Comment (mhio) [inline src/ip.py:234]: There a number of options/flags that could be passed in to `route` so I set it up for more use

File: src/ip.py [hunk 4]
```diff
@@ -185,45 +190,69 @@ def do_route_list(af):
...
     inet="-inet6 "
   if "dev"==via: gw = "-interface " + gw
-  return execute_cmd(SUDO + " " + ROUTE + " add " + inet + target + " " + gw)
+  return execute_cmd(SUDO + " " + ROUTE + " add " + inet + target + " " + gw + " " + options)
 
 def do_route_del(argv,af):
   target=argv[0]
   inet=""
+  if "blackhole"==target:
+    target=" ".join([ argv[1], "127.0.0.1", "-blackhole" ])
```
Comment (mhio) [inline src/ip.py:255]: Then I cheated here and didn't set it up for more options, as it was kind of sparse to start with

**Comment by brona** [comment]
Thank you for the contribution, since it is a bit more complex it might take me two weeks or so to review it and test it.

**Comment by brona** [comment]
I will merge it, however I will have to modify few things before releasing...

**Pull request merged**